### PR TITLE
updated to latest mesos-go, adds support for JSON MasterInfo in ZK

### LINKED ISSUE #228: Change in Mesos leader info in ZK may break Mesos-DNS
Mesos 0.24 will follow this convention for the leading master info in ZK: https://github.com/mesos/commons/blob/master/docs/leader_discovery.rst
https://issues.apache.org/jira/browse/MESOS-2340
This will break existing Mesos-DNS code which assumes the old protobuf. 

We need to update the code to handle correctly both 0.24 and older versions :(

### PR DESCRIPTION
**Pull request by jdef**
tested against mesos:tip, seems to do the trick

### CODE REVIEW

**Comment by jdef** [comment]
cc @cmaloney @nqn

**Comment by tsenart** [comment]
LGTM

File: Godeps/_workspace/src/github.com/mesos/mesos-go/mesosutil/constants.go [hunk 1]
```diff
@@ -2,5 +2,5 @@ package mesosutil
 
 const (
 	// MesosVersion indicates the supported mesos version.
-	MesosVersion = "0.20.0"
+	MesosVersion = "0.23.0"
```
Comment (nqn) [inline Godeps/_workspace/src/github.com/mesos/mesos-go/mesosutil/constants.go]: Don't want it to be 0.24.0?

Comment (tsenart) [inline Godeps/_workspace/src/github.com/mesos/mesos-go/mesosutil/constants.go]: We'll have to update that in mesos-go.

**Comment by tsenart** [comment]
Closing this to be re-opened with https://github.com/mesos/mesos-go/pull/178 vendored.

**Comment by tsenart** [comment]
I'll also add a CHANGELOG, tag things properly and do a Github release.

**Pull request closed without merging**